app/controllers/api/v1/spots_controller.rb

```diff
@@ -18,6 +19,26 @@ def destroy
       def spot_params
         params.require(:spot).permit(:title, :latitude, :longitude, :radius, :topic_id)
       end
+
+      def save_compatible_matches
+        find_compatible_spots
+        save_matches
+      end
+
+      def find_compatible_spots
+        @spots = Spot.within(@spot.radius, origin: [@spot.latitude, @spot.longitude])
+        @spots.reject { |spot| spot.user_id == current_user.id }
```

Comment:
Here you are filtering an array of objects in memory. This is really expensive if you have a a LOT of records. You should do this in the database rather than memory. 

Checkout this [guide](https://edgeguides.rubyonrails.org/active_record_querying.html). There you will find everything you need to pull from the database in the correct way.